homeassistant/components/homematic/climate.py

```diff
@@ -66,40 +60,54 @@ def temperature_unit(self):
         return TEMP_CELSIUS
 
     @property
-    def current_operation(self):
-        """Return current operation ie. heat, cool, idle."""
-        if HM_CONTROL_MODE not in self._data:
-            return None
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode.
 
-        # boost mode is active
-        if self._data.get('BOOST_MODE', False):
-            return STATE_BOOST
+        Need to be one of HVAC_MODE_*.
+        """
+        if "MANU_MODE" in self._hmdevice.ACTIONNODE:
+            if self._hm_controll_mode == self._hmdevice.MANU_MODE:
+                return HVAC_MODE_HEAT
+            return HVAC_MODE_AUTO
 
-        # HmIP uses the set_point_mode to say if its
-        # auto or manual
-        if HMIP_CONTROL_MODE in self._data:
-            code = self._data[HMIP_CONTROL_MODE]
-        # Other devices use the control_mode
-        else:
-            code = self._data['CONTROL_MODE']
+        # Simple devices
+        if self._data.get("BOOST_MODE"):
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_HEAT
```

Comment:
I don't know how those new modes apply here, but from the name it seems as if the boost mode corresponds to `HVAC_MODE_HEAT`, and `HVAC_MODE_AUTO` is the _regular_ mode.

Reply:
because `AUTO` means that you can control the temperature at this moment.

Reply:
wouldn't that be MODE_HEAT_COOL  ?